KOGITO-7421 kswf runtime tools quarkus dev ui docs (#38)
kiegroup/kogito-docs: serverlessworkflow/modules/ROOT/pages/tooling/quarkus-dev-ui-extension/quarkus-dev-ui-overview.adoc

= Kogito Serverless Workflow Tools - Quarkus Dev UI

> Quarkus applications comes with an interactive Dev UI, that provides developers with a number of tools to help in the process of development. You can plug in these tools to a Quarkus application in the form of extensions. The Kogito Serverless Workflow Tools is an extension, that provides a Console for you to view, manage and start Workflows Instances. You can add this extension as a _dependency_ to the _pom.xml_ of your Kogito Project.

Reviewer: ```suggestion
Quarkus applications come with an interactive Dev UI, that provides developers with a number of tools to help in the process of development. You can plug in these tools to a Quarkus application in the form of extensions. The Kogito Serverless Workflow Tools is an extension, that provides a Console for you to view, manage and start Workflows Instances. You can add this extension as a _dependency_ to the _pom.xml_ of your Kogito Project.
```

Please add a link to `Dev UI` documentation and `extensions` as well.

.Kogito Serverless Workflow Tools dependency in pom.xml
[source,xml]
----
<dependency>
  <groupId>org.kie.kogito</groupId>
  <artifactId>kogito-quarkus-serverless-workflow-devui</artifactId>
  <version>${VERSION}</version>
</dependency>
----

Another dependency for the Console is to add the _kogito-addons-quarkus-process-svg_ to provide the SVG diagram to the Consoles.

.Quarkus Addons:Process-SVG dependency in pom.xml
[source,xml]
----
<dependency>
  <groupId>org.kie.kogito</groupId>
  <artifactId>kogito-addons-quarkus-process-svg</artifactId>
</dependency>

> ----

Reviewer: We are using the `quarkus extensions add` approach to add dependencies. Then you can proceed and explain what it does. See: https://kiegroup.github.io/kogito-docs/serverlessworkflow/main/cloud/build-workflow-image-with-quarkus-cli.html#_using_a_kogito_example_application

Author: I have added as mentioned in the other doc.. Please let me know if that is okay

.The Quarkus application can be started by running the following command:
[source,shell]
----

> mvn clean quarkus:dev

Reviewer: We can use `quarkus dev` instead.

----

The Quarkus Dev UI can be accessed by using `http://localhost:8080/q/dev/`

.The Kogito Serverless Workflow Tools Extension in Quarkus Dev UI
image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-quarkus-dev-ui.png[]

> The Kogito Serverless Workflow Tools tab of the Dev UI shows the number of *Workflow Instances* available. Clicking on it will navitage you to the Console page.

Reviewer: ```suggestion
The Kogito Serverless Workflow Tools tab of the Dev UI shows the number of *Workflow Instances* available. Clicking on it will navigate you to the Console page.
```

.Kogito Serverless Workflow Tools dependency in pom.xml
[source,xml]
----
<dependency>
  <groupId>org.kie.kogito</groupId>
  <artifactId>kogito-quarkus-serverless-workflow-devui</artifactId>
  <version>${VERSION}</version>
</dependency>
----

Another dependency for the Console is to add the _kogito-addons-quarkus-process-svg_ to provide the SVG diagram to the Consoles.

.Command to install Kogito SVG add-on extension
[source,shell]
----
quarkus ext add org.kie.kogito:kogito-addons-quarkus-process-svg
----

Executing the command adds the following _dependency_ to the _pom.xml_ of your project.

.Quarkus Addons:Process-SVG dependency in pom.xml
[source,xml]
----
<dependency>
  <groupId>org.kie.kogito</groupId>
  <artifactId>kogito-addons-quarkus-process-svg</artifactId>
</dependency>
----

.The Quarkus application can be started by running the following command:
[source,shell]
----
quarkus dev
----

The Quarkus Dev UI can be accessed by using `http://localhost:8080/q/dev/`

.The Kogito Serverless Workflow Tools Extension in Quarkus Dev UI
image::tooling/quarkus-dev-ui-extension/kogito-swf-tools-quarkus-dev-ui.png[]

> The Kogito Serverless Workflow Tools tab of the Dev UI shows the number of *Workflow Instances* available. Clicking on it will navigate you to the Console page.

Reviewer 2: ```suggestion
The Kogito Serverless Workflow Tools card above shows the number of *Workflow Instances* available. Clicking on it will navigate you to the Workflow Instances page.
```

> .The Console has two tabs:

Reviewer 2: ```suggestion
.The extension has two pages:
```

* xref:tooling/quarkus-dev-ui-extension/quarkus-dev-ui-workflow-instances-page.adoc[Workflow  Instances]

> * xref:tooling/quarkus-dev-ui-extension/quarkus-dev-ui-workflow-definition-page.adoc[Workflow Definitions]

Reviewer: @AjayJagan, I think @hmanwani-rh can help us here since they should have writing guidelines for situations where we need to branch the main topic.

Author: sound good :+1:
